replica/table.cc

```diff
@@ -1180,7 +1180,11 @@ future<bool> table::perform_offstrategy_compaction() {
     // If the user calls trigger_offstrategy_compaction() to trigger
     // off-strategy explicitly, cancel the timeout based automatic trigger.
     _off_strategy_trigger.cancel();
-    return _compaction_manager.perform_offstrategy(as_table_state());
+    bool performed = false;
+    co_await for_each_compaction_group([this, &performed] (compaction_group& cg) -> future<> {
+        performed |= co_await _compaction_manager.perform_offstrategy(cg.as_table_state());
```

Comment:
Should be done in parallel, so the compaction manager is informed of all the tasks and can cancel them. As written this cannot be cancelled.